packages/dev/addons/src/lottie/lottie/animationParser.ts

```diff
@@ -0,0 +1,526 @@
+
+import type { AnimationConfiguration } from "../lottiePlayer";
+
+/**
+ * Type of the vector properties in the Lottie animation. It determines how the vector values are interpreted in Babylon.js.
+ */
+type VectorType = "Scale" | "Position" | "AnchorPoint";
+/**
+ * Type of the scalar properties in the Lottie animation. It determines how the scalar values are interpreted in Babylon.js.
+ */
+type ScalarType = "Rotation" | "Opacity";
+
+/**
+ * Default scale value for the scale property of a Lottie transform.
+ */
+const DefaultScale: IVector2Like = { x: 1, y: 1 };
+
+/**
+ * Default position value for the position property of a Lottie transform.
+ */
+const DefaultPosition: IVector2Like = { x: 0, y: 0 };
+
+/**
+ * Parses a lottie animation file and converts it into a format that can be rendered by Babylon.js
+ * Important: not all lottie features are supported, you can call .debug() after parsing an animation to see what features were not supported.
+ */
+export class AnimationParser {
+    private _packer: SpritePacker;
+    private readonly _renderingManager: RenderingManager;
+    private readonly _configuration: Required<AnimationConfiguration>;
+
+    private _unsupportedFeatures: string[];
+
+    private _parentNodes: Map<number, Node>; // Map of nodes to build the scenegraph from the animation layers
+    private _rootNodes: Node[]; // Array of root-level nodes in the animation, in top-down z order
+
+    // Loop variables to save allocations
+    private _shape: RawGraphicElement | undefined = undefined;
+
+    /**
+     * Creates a new instance of the Lottie animations parser.
+     * @param packer Object that packs the sprites from the animation into a texture atlas.
+     * @param renderingManager Object that manages the rendering of the sprites in the animation.
+     * @param configuration Configuration options for the animation parser.
+     */
+    public constructor(packer: SpritePacker, renderingManager: RenderingManager, configuration: Required<AnimationConfiguration>) {
+        this._packer = packer;
+        this._renderingManager = renderingManager;
+        this._configuration = configuration;
+
+        this._unsupportedFeatures = [];
+
+        this._parentNodes = new Map<number, Node>();
+        this._rootNodes = [];
+    }
+
+    /**
+     * Logs to the console all issues that were encountered during parsing the file.
+     */
+    public debug() {
+        for (let i = 0; i < this._unsupportedFeatures.length; i++) {
+            // eslint-disable-next-line no-console
+            console.log(this._unsupportedFeatures[i]);
+        }
+    }
+
+    /**
+     * Parses the Lottie file information so it can be rendered by Babylon.js
+     * @param fileContentAsJsonString The content of the lottie file as a JSON string.
+     * @returns An AnimationInfo instance containing the parsed animation data.
+     */
+    public loadFromData(fileContentAsJsonString: string): AnimationInfo {

```

Comment:
Is this expected to be called multiple times for a single instance of the parser?

Reply:
Good catch, this will actually fail as the sprite packer is single use. I need to address this.